client/image_build.go

```diff
@@ -94,6 +94,7 @@ func (cli *Client) imageBuildOptionsToQuery(options types.ImageBuildOptions) (ur
 	query.Set("cgroupparent", options.CgroupParent)
 	query.Set("shmsize", strconv.FormatInt(options.ShmSize, 10))
 	query.Set("dockerfile", options.Dockerfile)
+	query.Set("ignorefile", options.Ignorefile)
```

Comment:
Related to my previous comment, if we don't send this query parameter at all when the --ignore-file wasn't specified then the server should be able to ignore the "not found" error because it's defaulting.